pipeline/relation_embedder/batcher/src/main/scala/weco/pipeline/batcher/PathsProcessor.scala

```diff
@@ -0,0 +1,88 @@
+package weco.pipeline.batcher
+import grizzled.slf4j.Logging
+import org.apache.pekko.NotUsed
+import org.apache.pekko.stream.Materializer
+import org.apache.pekko.stream.scaladsl.{Sink, Source}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object PathsProcessor extends Logging {
+
+  /** Takes a list of strings, each representing a path to be processed by
+    * _downstream_
+    *
+    * This processor bundles the input paths together into Batches suitable for
+    * processing together, then passes those Batches on to _downstream_ for
+    * actual processing.
+    *
+    * @return
+    *   A sequence representing the positions within the input list of any paths
+    *   that were not successfully processed (That still seems a bit
+    *   SQS/SNS-driven. Should just be the actual failed paths, and the caller
+    *   should build a map to work it out if it wants to)
+    */
+  def apply(maxBatchSize: Int, paths: List[String], downstream: Downstream)(
+    implicit ec: ExecutionContext,
+    materializer: Materializer
+  ): Future[Seq[Long]] = {
+    generateBatches(maxBatchSize, paths)
+      .mapAsyncUnordered(10) {
+        case (batch, msgIndices) =>
+          Future {
+            downstream.notify(batch) match {
+              case Success(_) => None
+              case Failure(err) =>
+                error(s"Failed processing batch $batch with error: $err")
+                Some(msgIndices)
```

Comment:
This business with the indices is a relic of the SQS Stream driven version.  It's a bit tricky to untangle, so I've left it in place.

It might be easier to rewrite this later on when we no longer use that approach.